bash/drive-utils.bash

```diff
@@ -0,0 +1,529 @@
+                "${QUIET:-_print_center}" "justify" "${name}" " already exists." "=" && return 0
+            else
+                _print_center "justify" "Overwriting file.." "-"
+                _file_id="$(_json_value id 1 1 <<< "${existing_file_check_json}")"
+                clone_file_post_data="$(_drive_info "${_file_id}" "parents,writersCanShare" "${token}")"
+                if [[ ${_file_id} != "${file_id}" ]]; then
+                    curl --compressed -s \
+                        -X DELETE \
+                        -H "Authorization: Bearer ${token}" \
+                        "${API_URL}/drive/${API_VERSION}/files/${_file_id}?supportsAllDrives=true" &> /dev/null || :
+                    STRING="Updated"
+                else
+                    _collect_file_info "${existing_file_check_json}" || return 1
+                fi
+            fi
+        else
+            _print_center "justify" "Cloning file.." "-"
+        fi
+    else
+        _print_center "justify" "Cloning file.." "-"
+    fi
+
+    # shellcheck disable=SC2086 # Because unnecessary to another check because ${CURL_PROGRESS} won't be anything problematic.
+    clone_file_response="$(curl --compressed ${CURL_PROGRESS} \
+        -X POST \
+        -H "Authorization: Bearer ${token}" \
+        -H "Content-Type: application/json; charset=UTF-8" \
+        -d "${clone_file_post_data}" \
+        "${API_URL}/drive/${API_VERSION}/files/${file_id}/copy?supportsAllDrives=true" || :)"
+    for _ in 1 2 3; do _clear_line 1; done
+    if [[ -n ${clone_file_response} ]]; then
+        _collect_file_info "${clone_file_response}" || return 1
+        "${QUIET:-_print_center}" "justify" "${name} " "| ${readable_size} | ${STRING}" "="
+    else
+        _error_logging && return 1
+    fi
+    return 0
+}
+
+###################################################
+# Share a gdrive file/folder
+# Globals: 2 variables, 4 functions
+#   Variables - API_URL and API_VERSION
+#   Functions - _url_encode, _json_value, _print_center, _clear_line
+# Arguments: 3
+#   ${1} = gdrive ID of folder/file
+#   ${2} = Access Token
+#   ${3} = Email to which file will be shared ( optional )
+# Result: read description
+# Reference:
+#   https://developers.google.com/drive/api/v3/manage-sharing
+###################################################
+_share_id() {
+    [[ $# -lt 2 ]] && printf "%s: Missing arguments\n" "${FUNCNAME[0]}" && return 1
+    declare id="${1}" token="${2}" share_email="${3}" role="reader" type="anyone"
+    declare type share_post_data share_post_data share_response share_id
+
+    "${EXTRA_LOG:-:}" "justify" "Sharing.." "-" 1>&2
+
+    if [[ -n ${share_email} ]]; then
+        type="user"
+        share_post_data="{\"role\":\"${role}\",\"type\":\"${type}\",\"emailAddress\":\"${share_email}\"}"
+    else
+        share_post_data="{\"role\":\"${role}\",\"type\":\"${type}\"}"
+    fi
+
+    share_response="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)